test/extended/operators/olm.go

```diff
@@ -105,4 +105,19 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 		}
 	})
 
+	// OCP-21130 - [bug ALM-736]Fetching non-existent `PackageManifest` does not return 404
+	// author: bandrade@redhat.com
+	g.It("Fetching non-existent `PackageManifest` does not return 404", func() {
+		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "--all-namespaces", "--no-headers").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		packageserverLines := strings.Split(msg, "\n")
+		if len(packageserverLines) > 0 {
+			raw, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "a_package_that_not_exists", "-o yaml", "--loglevel=8").Output()
+			o.Expect(err).To(o.HaveOccurred())
+			o.Expect(raw).To(o.ContainSubstring("\"code\": 404"))
+		} else {
+			e2e.Failf("No packages to evaluate if 404 works when a package does not exists")
```

Comment:
Even if there are no packages I would expect the same behavior when calling `oc get` on a package that doesn't exist. I don't think this failure case is necessary for this test.

Reply:
Yes, make sense. But, if there are not any packages, this may be another issue. But, we should report it in the test. What do you think?

Reply:
Please check https://github.com/openshift/origin/pull/23772#discussion_r356960948 , I had the same discussion with @bparees about throwing failure by default.